src/fights.ts

```diff
@@ -351,6 +358,26 @@ function pygmyOptions(forceEquip: Item[] = []) {
   };
 }
 
+let _bestShadowRift: Location | null = null;
+export function bestShadowRift(): Location {
+  if (!_bestShadowRift) {
+    _bestShadowRift =
+      chooseRift({
+        canAdventure: true,
+        sortBy: (l: Location) => {
+          const drops = getMonsters(l)
+            .map((m) => Object.keys(itemDrops(m)).map((s) => toItem(s)))
+            .reduce((acc, val) => acc.concat(val), []);
+          return sum(drops, garboValue);
+        },
+      }) ?? $location.none;
+    if (_bestShadowRift === $location.none) {
+      throw new Error("Failed to find a suitable Shadow Rift to adventure in");
+    }
+  }
+  return myInebriety() > inebrietyLimit() ? $location`Shadow Rift` : _bestShadowRift;
```

Comment:
I don't understand the intention of this

Reply:
![image](https://user-images.githubusercontent.com/98746573/227933572-2b7da904-66e5-48e3-8dc8-970384702ebd.png)
It's to address a silly bug

Reply:
Then put a comment to address that